javav2/example_code/transcribe-streaming/src/main/java/com/amazonaws/transcribestreaming/TranscribeStreamingDemoApp.java

```diff
@@ -206,0 +1,205 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.amazonaws.transcribestreaming;
+
+import org.reactivestreams.Publisher;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.transcribestreaming.TranscribeStreamingAsyncClient;
+import software.amazon.awssdk.services.transcribestreaming.model.*;
+import javax.sound.sampled.*;
+import java.io.*;
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicLong;
+
+// snippet-start:[transcribe.java-streaming-demo]
+public class TranscribeStreamingDemoApp {
+    private static final Region REGION = Region.US_EAST_1;
+    private static TranscribeStreamingAsyncClient client;
+
+    public static void main(String args[])
+            throws URISyntaxException, ExecutionException, InterruptedException, LineUnavailableException {
+
+        client = TranscribeStreamingAsyncClient.builder()
+                .credentialsProvider(getCredentials())
+                .region(REGION)
+                .build();
+
+        CompletableFuture<Void> result = client.startStreamTranscription(getRequest(16_000),
+                new AudioStreamPublisher(getStreamFromMic()),
+                getResponseHandler());
+
+        result.get();
+        client.close();
+    }
+
+    private static InputStream getStreamFromMic() throws LineUnavailableException {
+
+        // Signed PCM AudioFormat with 16kHz, 16 bit sample size, mono
+        int sampleRate = 16000;
+        AudioFormat format = new AudioFormat(sampleRate, 16, 1, true, false);
+        DataLine.Info info = new DataLine.Info(TargetDataLine.class, format);
+
+        if (!AudioSystem.isLineSupported(info)) {
+            System.out.println("Line not supported");
+            System.exit(0);
+        }
+
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *line*. Currently, there are execution paths that do not contain closure statements, for example, when TargetDataLine.open throws an exception. Ensure *line* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).